File: integer-to-roman/integer-to-roman.java
class Solution {
    public String intToRoman(int num) {
        Roman[] list = new Roman[] {
            new Roman("M", 1000),
            new Roman("CM", 900),
            new Roman("D", 500),
            new Roman("CD", 400),
            new Roman("C", 100),
            new Roman("XC", 90),
            new Roman("L", 50),
            new Roman("XL", 40),
            new Roman("X", 10),
            new Roman("IX", 9),
            new Roman("V", 5),
            new Roman("IV", 4),
            new Roman("I", 1)
        };
        
        String result = "";
        int pointer = 0;
        
        while(num > 0) {
            while (list[pointer].numericValue > num) pointer++;
            result += list[pointer].value;
            num -= list[pointer].numericValue;
        }
        
        return result;
    }
    
    class Roman {
        public String value = "";
        public int numericValue = 0;
        
        public Roman(String value, int numericValue) {
            this.value = value;
            this.numericValue = numericValue;
        }
    }
}

File: integer-to-roman/integer-to-roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        map = [
            ('I', 1),
            ('IV', 4),
            ('V', 5),
            ('IX', 9),
            ('X', 10),
            ('XL', 40),
            ('L', 50),
            ('XC', 90),
            ('C', 100),
            ('CD', 400),
            ('D', 500),
            ('CM', 900),
            ('M', 1000)
        ]
        
        
        p = len(map) - 1
        res = ''
        
        while num:
            while map[p][1] > num:
                p -= 1
            res += map[p][0]
            num -= map[p][1]
            
        return res